tests/node.py

```diff
@@ -0,0 +1,118 @@
+import os
+from pathlib import Path
+from subprocess import STDOUT, Popen, run
+
+from .hopr import HoprdAPI
+
+
+class Node:
+    def __init__(
+        self,
+        api_port: int,
+        p2p_prt: int,
+        api_token: str,
+        dir: Path,
+        host_addr: str,
+        network: str,
+        cfg_file: Path = None,
+    ):
+        # initialized
+        self.api_port: int = api_port
+        self.p2p_port: int = p2p_prt
+        self.dir: Path = dir
+        self.host_addr: str = host_addr
+        self.api_token: str = api_token
+        self.network: str = network
+
+        # optional
+        self.cfg_file: Path = cfg_file
+
+        # generated
+        self.safe_address: str = None
+        self.module_address: str = None
+        self.proc: Popen = None
+
+        # private
+        self.peer_id: str = None
+        self.address: str = None
+
+    @property
+    def api(self):
+        return HoprdAPI(f"http://{self.host_addr}:{self.api_port}", self.api_token)
+
+    def create_local_safe(self, custom_env: dict):
+        res = run(
+            [
+                "hopli",
+                "create-safe-module",
+                "--network",
+                self.network,
+                "--identity-from-path",
+                f"{self.dir}.id",
+                "--contracts-root",
+                "./ethereum/contracts",
+                "--hopr-amount",
+                "20000.0",
+            ],
+            env=os.environ | custom_env,
+            check=True,
+            capture_output=True,
+            text=True,
+        )
+
+        for el in res.stdout.split("\n"):
+            if el.startswith("safe: address 0x"):
+                self.safe_address = el.split()[-1]
+
+            if el.startswith("module: address 0x"):
+                self.module_address = el.split()[-1]
+
+        return self.address is not None and self.module_address is not None
+
+    def setup_node(self, password: str, config_file: Path, dir: Path):
```

Comment:
```suggestion
    def setup(self, password: str, config_file: Path, dir: Path):
```

Reply:
Setup and cleanu should be `_setup` and `_cleanup` and instead the node should offer a context manager.

Reply:
If you are up to this amount of fun, you can go a set this up yourself. I am not sure it is less error prone, as functionality would be the same: nodes are already yielded and then cleanup when test finishes. Also, if a user interrupts the test manually, the cleanup would not be done (same as now). Only the number of lines in the `swarm7` fixture would be smaller. In addition, yielding a node right after setting it up (call setup, safe_setup, wait for api to be ready) would mean some setup logs in between test logs, as node would be only setup on request, not initially and then used randomly.